packages/execution-environments/src/common/endowments/network.ts

```diff
@@ -257,14 +276,50 @@ const createNetwork = () => {
       return onClosed;
     }
 
+    #createWrapped(listener: WebSocketCallback): WebSocketCallback;
+
+    #createWrapped(listener: null): null;
+
+    #createWrapped(
+      listener: WebSocketCallback | null,
+    ): WebSocketCallback | null;
+
+    #createWrapped(
+      listener: WebSocketCallback | null,
+    ): WebSocketCallback | null {
+      if (listener === null) {
+        return null;
+      }
+
+      return (e) => {
+        listener.apply(this, [
+          {
+            ...e,
```

Comment:
I'm not sure if methods like `preventDefault()` are of any use on WebSocket events, but I believe spreading will effectively break them.